### ISSUE TITLE
Black bar above keyboard in Support SDK

### ISSUE THREAD

**Issue by arorapulkit** [zendesk/zendesk_sdk_ios#283]
- SDK version: 1.7.6.1
- Xcode version: 7.3.1
- iOS or Simulator: Both
- iOS Version: 9.0+
- Swift version: 2.3.1

Steps to Reproduce:

1. Open Tickets
2. Tap on a ticket
3. Tap on comments textfield to add a comment
<img width="487" alt="screen shot 2017-03-23 at 3 06 02 am" src="https://cloud.githubusercontent.com/assets/14070166/24222956/fbff71bc-0f79-11e7-8cac-bc3ee15cf2fc.png">

**Comment by baz8080** [comment]
Hi @arorapulkit. Please could you try this on the latest version of the Support SDK (v1.9.1.1) and let me know if you are having the same problem? I'm running the latest version of the Support SDK on iOS 10.2.1 and am unable to reproduce this.

**Comment by RonanMcH** [comment]
Hi @arorapulkit , 
As we haven't heard from you in a while, I am going to tentatively close this github issue. If the problem still persists for you, feel free to reopen it.
Thanks,
Ronan.

**Comment by haydenholligan** [comment]
This is happening to me too for 1.9.2.1.

**Comment by haydenholligan** [comment]
I don't want to go through the trouble of creating an entirely new project, but this is the simplest I could get in my current project. 

New view controller, embedded in navigation controller. 

`        if let request = requests?[0],
            let commentsVC = ZDKCommentsViewController(request: request) {
            self.navigationController?.pushViewController(commentsVC, animated: true)
        }
`

Then selecting the textfield at the bottom to open the keyboard will open the keyboard with the black bar as shown above.

**Comment by RonanMcH** [comment]
Hi @haydenholligan ,
I still can't reproduce this, even recreating a sample application and using the code you supplied and the conditions you specified. Were testing on physical device or simulator? Is there anymore code unique to your project that you could share and help us debug this issue?
Thanks,
Ronan.

**Comment by haydenholligan** [comment]
Both were on real devices. I'll look into it tonight and see if I can reproduce it.

**Comment by StevenDiviney** [comment]
@haydenholligan Are you presenting the SDK from a controller with a tab bar? All of our "pushController" calls can be passed a layout guide to correct for this.

https://github.com/zendesk/zendesk_sdk_ios/blob/master/ZendeskSDK.framework/Headers/ZDKRequests.h#L146

**Comment by haydenholligan** [comment]
I am using a tab bar in the main app. I created my own UITableView and am grabbing requests with `ZDKRequestProvider`. When a user clicks on a ticket (TableViewCell), I get the correlated request and create a `ZDKCommentsViewController`. When I tried replicating the problem in the same project, the navigation stack was simply a UINavigationController and a UIViewController. I just tried `ZDKRequests.pushRequestList` and `ZDKRequests.presentRequestList` and clicking on a ticket and bringing up the keyboard still presents this problem. I don't think a layout guide is what I need.

**Comment by haydenholligan** [comment]
I think it might be a problem with toolbars. I was using a framework that adds a toolbar to all keyboards when they show. It doesn't affect other views, so the chat view conroller might be compensating or something

**Comment by haydenholligan** [comment]
The easiest way I can reproduce it is to add IQKeyboardManager (https://github.com/hackiftekhar/IQKeyboardManager) to the SampleApp project. Even with the toolbar off it looks like the view is over-compensating. I can't debug in because your source code is hidden.

**Comment by tecknut** [comment]
@haydenholligan Thanks, we're going to add IQKeyboardManager and try reproduce this.

**Comment by tecknut** [comment]
@haydenholligan Could you try disabling IQKeyboardManager for this class `ZDKCommentInputViewController`? Seems like other people have had issues with IQKeyboard. We're still looking into this to see if there is something we can do to retain IQKeyboard on the comments controller, but if this works for you in the mean time it might be worth a try. 

https://github.com/hackiftekhar/IQKeyboardManager/issues/252#issuecomment-280608646

**Comment by haydenholligan** [comment]
Did that work for you? I just tried with no success. 

Actually, I just disabled IQKeyboardManager entirely at the start of my AppDelegate, did a clean and fresh app install, and the toolbar is still showing up. I tried this in the SampleApp and `IQKeyboardManager.shared().isEnabled = false` removed the bug, but `IQKeyboardManager.shared().isEnableAutoToolbar = false` didn't. 

I noticed that in the SampleApp, when the bug occurs the entire view including the navigation bar shifts up. My app, the navigation bar doesn't. I'm not sure what's going on.

**Comment by oarrabi** [comment]
Hello @haydenholligan,

> I noticed that in the SampleApp, when the bug occurs the entire view including the navigation bar shifts up. My app, the navigation bar doesn't. I'm not sure what's going on.
This is the issue we are noticing in our sample app. We are still not able to replicate the black bar issue in our tests.
We added this bug to our backlog for the issue of the pushed navigation bar but we can't do more about the black bar since we cannot replicate it. 

Thanks,
Omar...

**Comment by haydenholligan** [comment]
Has the bug in the backlog been fixed yet? I have removed IQKeyboardManager from the project and still getting the black bar bug. I have a hunch they're related, but can't know for sure until your bug has been fixed. It's been over 2 months and this app is supposed to be live next week.

**Comment by baz8080** [comment]
Hi @haydenholligan 

The bug is still in our backlog. They may be related, but it's difficult to know for sure because we were never able to reproduce the black bar issue. Do you have a small sample application which shows the problem that you could share with us?

**Comment by haydenholligan** [comment]
It's been over 4 months since discovering the shifting bug @nsomar mentioned in April. Has it been fixed yet? I still don't see a point in trying to debug the black bar before fixing the possibly/probably related issue.

**Comment by RonanMcH** [comment]
Hi @haydenholligan ,
I have spent a bit of time investing this issue, this morning.

* I tried integrating `IQKeyboardManager` into our sample and enabling it in the `appdelegate`with `[IQKeyboardManager sharedManager].enable = YES;`. I could see the bug.
* I disabled the Keyboard managing distance in the appdelegate then with this code `[IQKeyboardManager sharedManager].enable = NO;` and the keyboard remained but the bug disappeared. 

Outside of this instance, I haven't been able to recreate this bug. In order to debug it any further, we need some information about your nav stack and project setup. A sample project would be the quickest way to resolve it as it seems to be unique to your project.
Thanks,
Ronan
